src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,436 @@
+package duke;
+
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.List;
+import java.util.Optional;
+import java.util.Scanner;
+
+/**
+ * Represents Duke itself, the Chat bot.
+ */
+public class Duke {
+
+    private TaskList tasks;
+    private final Storage storage;
+    private final Ui ui;
+
+    /**
+     * Constructor for Duke.
+     *
+     * @param filePath File path to store/retrieve previous task list.
+     */
+    public Duke(String filePath) {
+        storage = new Storage(filePath);
+        tasks = new TaskList(storage.load());
+        ui = new Ui();
+    }
+
+    /**
+     * Adds the current TaskList into a specified file for storage purposes.
+     *
+     * @param taskList List of Tasks to be saved.
+     * @throws IOException If specified file does not exist.
+     */
+    public void saveTaskList(List<Task> taskList) throws IOException {
+        FileWriter fw = new FileWriter(storage.getFilePath());
+        for (Task a : taskList) {
+            fw.write(a.getDataToSave() + System.lineSeparator());
+        }
+        fw.close();
+    }
+
+    /**
+     * Returns the response from Duke based on user input.
+     *
+     * @param userInput User input.
+     * @return Respnose from Duke.
+     */
+    public String getResponse(String userInput) {
```

Comment:
Instead to aggregating all event handles (both parsing and UI response) in one method, would it not be better to add separate methods to separately handle separate commands?